source/dagger/gameplay/ping_pong/player_scores.cpp

```diff
@@ -50,6 +50,12 @@ void PlayerScoresSystem::Run()
     if (ballOnField == 0)
     {
         // TODO: set speed to be random in both directions
-        CreatePingPongBall(s_TileSize, ColorRGBA(1, 1, 1, 1), { rand()%10 + 4,rand()%10 + 4,0 },   { 0,rand()%(s_FieldHeight / 2),0 });
+        float xSpeed = rand() % 17 - 8;
+        while ((xSpeed > -3) && (xSpeed < 3))
+        {
+            xSpeed = rand() % 17 - 8;
+        }
```

Comment:
You can use `std::clamp` instead. https://en.cppreference.com/w/cpp/algorithm/clamp

Reply:
So basically, `std::clamp(rand(), -3, 3)` should do the trick.

Reply:
Also, take into account what happens if your value here ends up being 0...

Reply:
Alright, thank you for reviewing and for the helpful suggestion. Cheers!